### PR TITLE
Implemented deleted button and removed upgrade button

### PR DESCRIPTION
**Pull request by KasperDamgaard**

### CODE REVIEW

**Comment by MadsSvejstrup** [comment]
Deleted datasets still show up under relations and errors are thrown if i try to click on it
image

File: src/app/pages/details-page/details-page.component.ts [hunk 1]
```diff
@@ -60,85 +62,27 @@ export class DetailsPageComponent implements OnInit, OnDestroy {
...
-    this.dataSet.sourceTransformation = new Transformation ({
-      id: '',
-      sourceDatasets: [],
-      createdDate: new Date(),
-      modifiedDate: new Date()
+  deleteDataset() {
+    const confirmDialog = this.dialog.open(ConfirmationDialogComponent, {
+      data: {
+        title: 'Confirm dataset deletion',
+        message: 'You are about to delete the dataset ' + this.dataSet.name
```
Comment (MadsSvejstrup) [inline src/app/pages/details-page/details-page.component.ts:75]: These should probably be provided by TranslateService, but it doesn't make a huge difference right now as we only supports english

File: src/app/pages/details-page/details-page.component.ts [hunk 2]
```diff
@@ -60,85 +62,27 @@ export class DetailsPageComponent implements OnInit, OnDestroy {
...
-    // });
-  }
-
   // Navigate to edit and send dataset data to
   editDataSet() {
     this.dataStewardHandlerService.setDataSet(this.dataSet);
     this.router.navigate(['/datasteward']);
   }
 
-  promoteDatasetToStock() {
```
Comment (MadsSvejstrup) [inline src/app/pages/details-page/details-page.component.ts:90]: Are we completely sure that we can remove this functionality?

Comment (KasperDamgaard) [reply src/app/pages/details-page/details-page.component.ts:90]: Well, no I guess.. Just 95%. Let's confirm with Sune on monday

**Comment by KasperDamgaard** [comment]
> Deleted datasets still show up under relations and errors are thrown if i try to click on it (asdfasdf has been deleted)
> image

Good catch - I'll look into that

**Pull request merged**